libs/googlePayments.js

```diff
@@ -0,0 +1,119 @@
+const nconf = require('nconf');
+const iap = require('in-app-purchase');
+const request = require('request');
+const USERS_BATCH = 10;
+const BASE_URL = nconf.get('BASE_URL');
+const subscriptions = require('../libs/subscriptions');
+
+function cancelSubscriptionForUser (user) {
+  return new Promise((resolve, reject) => {
+    request({
+      url: `${BASE_URL}/iap/android/subscribe/cancel`,
+      method: 'GET',
+      qs: {
+        noRedirect: 'true',
+        _id: user._id,
+        apiToken: user.apiToken,
+      },
+    }, (habitError, habitResponse, body) => {
+      if (!habitError && habitResponse.statusCode === 200) {
+        return resolve();
+      }
+
+      reject(habitError || body); // if there's an error or response.statucCode !== 200
+    });
+  });
+}
+
+function scheduleNextCheckForUser (habitrpgUsers, user, subscription, nextScheduledCheck) {
+  if (subscription.expirationDate < nextScheduledCheck) {
+    nextScheduledCheck = subscription.expirationDate;
+  }
+  return habitrpgUsers.update(
+    {
+      _id: user._id,
+    },
+    {
+      $set: {
+        'purchased.plan.nextPaymentProcessing': nextScheduledCheck,
+        'purchased.plan.nextBillingDate': subscription.expirationDate,
+      },
+    }, e => {
+      if (e) return reject(e);
```

Comment:
the callback here is not necessary because habitrpgUsers.update returns a promise. Also this is not inside `new Promise` so resolve and reject aren't defined